### PR TITLE
feat(adac-vscode): added vscode extension & updated relase script to create vscode extesion release

### PR DESCRIPTION
**Pull request by yogeshsmindfire**
This PR implements the comprehensive VS Code extension for ADAC, enforces strict TypeScript compliance across all 23 workspace packages, and integrates critical security enhancements for the web server and UI.

---

### 🔹 VS Code Extension (`@mindfiredigital/adac-vscode`)
*   **IntelliSense & Autocomplete**: Full schema-aware autocomplete for ADAC YAML files using structured schema data.
*   **Live Validation**: Real-time diagnostic provider for syntax and logical errors (e.g., non-unique IDs).
*   **Diagram Preview**: Command to preview the generated diagram directly within a VS Code Webview.
*   **Formatting**: Built-in document formatter for standardized YAML structure.
*   **Snippets**: Pre-defined snippets for rapid cloud architecture drafting (EC2, S3, VPC, etc.).

### 🔹 Modern Web UI Enhancements
*   **Custom Toaster**: Replaced legacy `alert()` calls with a premium, glassmorphic `Toaster` component for non-blocking notifications.
*   **Strict UI Typing**: Refactored `Flow` components and configuration drawers to use strict TypeScript interfaces from `@mindfiredigital/adac-schema`.

---

### 🔒 Web Server Security
*   **Helmet.js Integration**: Implemented security-focused HTTP headers to protect against common web vulnerabilities.
*   **XSS Protection**: Added server-side sanitization for all diagram metadata and configuration inputs using the `xss` library.
*   **API Validation**: Standardized error handling and added unit tests for all REST API endpoints.

### 💎 Type-Safety (Zero `any` Policy)
*   **Global Refactor**: Systematically eliminated the `any` type across the entire monorepo (Core, Parser, Layout, Web, etc.).
*   **Generic Optimization**: Refactored `@mindfiredigital/adac-optimizer` to use type-safe generics instead of loose object types.
*   **AJV Schema Hardening**: Resolved AJV constructor type conflicts in the `schema` package to ensure strict type checking during validation.

---

### 🛠️ Infrastructure Improvements
*   **Standardized Metadata**: Updated all `package.json` files with consistent author ("Mindfire Digital"), keywords, and repository information.
*   **Compliance Docs**: Added `CONTRIBUTING.md`, `CHANGELOG.md`, and `CODE_OF_CONDUCT.md` to the root directory.
*   **Automated JS Docs**: Generated comprehensive JSDoc documentation for all core functions across the workspace to improve developer experience and IDE suggestions.

### 🏗️ Build & CI/CD
*   **Unified Build Pipeline**: Verified `pnpm run build` stability across all packages.
*   **Security Auditing**: Integrated `pnpm audit` into the CI workflow to block high/critical vulnerabilities.
*   **Lockfile Synchronization**: Repaired and synchronized `pnpm-lock.yaml` and `package-lock.json` to ensure deterministic builds.

---

## 🧪 Testing & Quality
*   **100+ Unit Tests**: Refactored existing test suites (Vitest) to comply with new strict types.
*   **Extension Mocks**: Implemented complex VS Code environment mocks for unit testing providers without a running VS Code instance.
*   **Coverage Pass**: All tests in `packages/core`, `packages/web`, and `packages/vscode` are passing with **Exit Code 0**.

---

## ⚠️ Breaking Changes
*   **None**. All changes are additive or internal refactors that maintain backward compatibility with existing ADAC YAML schemas.

## ✅ Checklist
- [x] All `any` types replaced with strict interfaces.
- [x] `pnpm run build` and `pnpm run test` passing globally.
- [x] Security headers and XSS checks implemented.
- [x] Documentation and JSDocs updated.
- [x] VS Code extension verified in local development.

- Architecture / Package API changes
  - New VS Code extension package @mindfiredigital/adac-vscode (packages/vscode/*): full extension manifest (activationEvents, contributes: language, grammar, snippets, commands, menus, configuration), MIT LICENSE, .vscodeignore, strict tsconfig, and packaging/build scripts.
  - Extension runtime wiring: activate/deactivate entrypoints; registered DiagnosticProvider, CompletionProvider, HoverProvider; commands adac.validate, adac.formatDocument, adac.previewDiagram; status bar item; isAdacDocument utility.
  - Schema/metadata exports for IntelliSense: AWS_SERVICES, GCP_SERVICES, ALL_SERVICES and suggestion lists (CONNECTION_TYPES, PROTOCOLS, APPLICATION_TYPES, ENVIRONMENTS, CLOUD_PROVIDERS, TIERS, COMPLIANCE_FRAMEWORKS, LAYOUT_ENGINES, AWS_REGIONS, GCP_REGIONS, PRICING_MODELS).
  - New TypeScript interfaces for extension types: ServiceInfo, AdacDiagnostic, KeyPathContext.

- Core Logic / Generators
  - DiagnosticProvider: YAML parse error diagnostics and ADAC structural validation (required top-level/metadata fields, metadata.created warning, clouds/services validations), duplicate-ID detection across applications/clouds/services/connections, unknown connection reference warnings, debounced validation and proper disposal.
  - CompletionProvider: context-aware completions (top-level snippets, service names, provider-aware region suggestions, enums, environment/tier/layout/pricing, ID/reference completions for connections.from/to and runs, service property key snippets).
  - HoverProvider: Markdown hovers for services (name/description/category/provider), connection-type descriptions, compliance-framework info, top-level key docs, and ID reference listings with line context.
  - Commands:
    - adac.formatDocument: parses with js-yaml and rewrites YAML with deterministic formatting (2-space indent, 120 cols, no refs, single-quote handling).
    - adac.previewDiagram: Webview-based SVG preview via @mindfiredigital/adac-core.generateDiagramSvg with monorepo-root asset resolution, render-sequencing guards, debounced live updates (800ms), and safe error/log pages.

- Web UI / Components
  - VS Code Webview diagram preview embedding generated SVG plus generation duration and logs; web UI change replacing alert() with a glassmorphic Toaster.
  - Language/editor support: language-configuration.json (comments, brackets, auto-close, indentation rules, wordPattern) and TextMate grammar adac.tmLanguage.json for ADAC-specific highlighting.
  - Snippets: ~11 curated ADAC YAML snippets (scaffold, AWS/GCP service templates, connections, applications, AWS cloud, VPC, ECS Fargate, RDS, cost, monitoring, metadata) with placeholders and date tokens.
  - Status bar: right-aligned $(cloud) ADAC item visible for ADAC documents.

- Tooling / Docs
  - Extension build & packaging: esbuild-based bundler (packages/vscode/esbuild.js), strict tsconfig, packaging with @vscode/vsce to produce .vsix; pack-release updated to build/package the VSIX and include it in releases/.
  - Release & CI: GitHub Actions release workflow updated (contents: write), job renamed to “Build, Publish & Upload Release Assets”, runs node scripts/pack-release.js, uploads releases/ artifact (90d retention) and conditionally attaches artifacts on release events; release summary step lists artifacts with sizes. CI tweak: coverage upload now gated on node-version == 20.
  - scripts/pack-release.js: switched to pnpm pack --json to locate tarballs, builds VSIX, moves artifacts into releases/, and prints an artifacts summary.
  - Monorepo hygiene & docs: added CONTRIBUTING.md, CHANGELOG.md, CODE_OF_CONDUCT.md, generated JSDoc, standardized package.json metadata, synchronized lockfiles, and expanded commitlint type-enum.
  - Security & hardening: Helmet.js for security headers, server-side XSS sanitization (xss) for diagram metadata/config inputs, standardized API error handling, and unit tests for REST endpoints.
  - Type-safety & refactors: removed all any types across 23 packages, refactored adac-optimizer to use generics, resolved AJV constructor type conflicts.
  - Tests & quality integration: 100+ unit test updates for strict types; comprehensive Vitest tests for the VS Code extension (mocks) covering schema data, diagnostics, completions, hovers, commands, edge cases; relevant package tests passing; pnpm audit integrated into CI.

- Breaking changes
  - None

### CODE REVIEW

**Comment by uzerjamal-mfsi** [comment]
@coderabbitai review

**Comment by uzerjamal-mfsi** [comment]
@coderabbitai full review

**Comment by uzerjamal-mfsi** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai full review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

**Comment by yogeshsmindfire** [comment]
@coderabbitai review

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,12 +41,45 @@ jobs:
...
       - name: Build all packages
         run: pnpm run build
 
       - name: Run tests
         run: pnpm run test
 
+      # ── Package artifacts on Node 22 only ────────────────
+      - name: Pack npm packages
+        if: matrix.node-version == 22
+        run: |
```
[the remark this answers is not in the document]
Comment (yogeshsmindfire) [reply .github/workflows/ci.yml]: Restored 22 to the matrix

**Comment by yogeshsmindfire** [comment]
@coderabbitai full review

**Comment by yogeshsmindfire** [comment]
@coderabbitai full review

**Comment by yogeshsmindfire** [comment]
@coderabbitai full review
